docs/ref/forms/widgets.txt

```diff
@@ -195,6 +195,18 @@ Django will then include the extra attributes in the rendered output:
 You can also set the HTML ``id`` using :attr:`~Widget.attrs`. See
 :attr:`BoundField.id_for_label` for an example.
 
+For widgets that have options on their own, like those inheriting from ``ChoiceWidget``,
+you can also pass option-specific attributes using ``option_attrs``.
+For example, if you wish to set custom CSS classes for a RadioSelect's options, you could
+do as follows:
+
+.. code-block:: pycon
+
+    >>> from django.forms.widgets import RadioSelect
+    >>> widget = RadioSelect(choices=(("J", "John"),), option_attrs={"class": "special"})
+    >>> widget.render(name="beatle", value=["J"])
+    '<div><div>\n    <label><input type="radio" name="beatle" value="J" class="special" checked>\n John</label>\n\n</div>\n</div>'
+
```

Comment:
I am debating about whether this section makes most sense here or in [widgets inheriting from Select](https://docs.djangoproject.com/en/5.0/ref/forms/widgets/#widgets-inheriting-from-the-select-widget). If it remains here, it might be nice to link it

This also needs a `versionchanged:: 5.2` note to state that `option_attrs` is new